### PR TITLE
CSV-217 add flush setting to CSVFormat and create close method with flush setting

### PR DESCRIPTION
**Pull request by alex270295**

### CODE REVIEW

**Comment by alex270295** [comment]
Haven't got any good ideas to test `flush` method here. Maybe except this one:
mock `Appendable out` in `CSVPrinter` and verify that flush method was invoked or not depends on flush property value. 

Any suggestions?

**Comment by garydgregory** [comment]
Mocking is likely the only way to go here to make sure the API is called or
not.

Gary

**Comment by alex270295** [comment]
Maybe we should protect or explicitly document behavoiur if both `org.apache.commons.csv.CSVPrinter#close(boolean)`  `org.apache.commons.csv.CSVFormat#withAutoFlush` 
used at same time. Now it will flush if at least one was true.

**Comment by alex270295** [comment]
it seems the new method `org.apache.commons.csv.CSVPrinter#close(boolean)` is not such a cool idea, very badly combined with try-with-resources

File: src/main/java/org/apache/commons/csv/CSVPrinter.java [hunk 1]
```diff
@@ -81,6 +81,17 @@ public CSVPrinter(final Appendable out, final CSVFormat format) throws IOExcepti
 
     @Override
     public void close() throws IOException {
+        close(false);
+    }
+
+    public void close(boolean flush) throws IOException {
+        if (flush || format.getAutoFlush()) {
+            System.out.println(flush);
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/csv/CSVPrinter.java]: Don't need those println()'s ;-)

Comment (alex270295) [reply src/main/java/org/apache/commons/csv/CSVPrinter.java]: sry, debug =)

**Pull request closed without merging**